### PR TITLE
Configurable csv delimiter in aggregate script, updated pandas and case insensitive subscriber comparison

### PR DESCRIPTION
**Pull request by stankolubomir**
- configurable csv delimiter in aggregate script to be able to change the delimiter with the script argument
- updated pandas dependency - makes DataFrame more resilient to invalid data types
- subscriber string comparison is now case insensitive - fixes possible missing data in aggregation

### CODE REVIEW

File: aggregate/utils/pageviews.py [hunk 1]
```diff
@@ -238,32 +238,35 @@ def __init__(self):
...
+                                'purchase': 0,
+                                'refund': 0
+                            }
+
+                        if row['step'] not in ['checkout', 'payment', 'purchase', 'refund']:
+                            raise Exception(
+                                "unknown commerce step: " + row['step'] + ' for browser_id: ' + row['browser_id'])
+                        else:
+                            self.browser_commerce_steps[row['browser_id']][row['step']] += 1
+                except KeyError:
```
Comment (rootpd) [inline aggregate/utils/pageviews.py]: When can this KeyError happen? Why do we silently let it continue instead of failing?

Comment (stankolubomir) [reply aggregate/utils/pageviews.py]: When the browser_id is empty in exported csv, which can happen during testing, then the aggregation thwors an exception and does not finish. Then it is not possible to test the whole process of aggregation. This is a workaround only for an empty browser_id. There should not be an empty browser_id in the production export. So this won't matter.

Comment (stankolubomir) [reply aggregate/utils/pageviews.py]: There is now configurable option to ignore empty browser id only when appropriate cmd line argument is passed.

**Comment by rootpd** [comment]
Thanks, merged via https://github.com/remp2020/pythia-tools/commit/09a68cf77bd732e99188502bab2f35f27a23b8fd.

**Pull request closed without merging**